### PR TITLE
gcp bigquery detections

### PR DESCRIPTION
**Pull request by calkim-panther**
### Background

GCP Large Bigquery Scan Detection
GCP destructive bigquery detection
GCP Logging Settings Modified

### CODE REVIEW

File: rules/gcp_audit_rules/gcp_destructive_queries.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+from panther_base_helpers import deep_get
+
+DESTRUCTIVE_STATEMENTS = ("UPDATE", "DELETE", "DROP_TABLE", "ALTER_TABLE", "TRUNCATE_TABLE")
```
Comment (edyesed) [inline rules/gcp_audit_rules/gcp_destructive_queries.py:3]: it seems weird to me to do  `item in tuple`. Could this be retyped to either list or set?

File: rules/gcp_audit_rules/gcp_destructive_queries.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def rule(event):
+    return any(
+        [
+            all(
```
Comment (edyesed) [inline rules/gcp_audit_rules/gcp_destructive_queries.py:9]: I would recommend breaking this out into two conditionals. I am having some trouble creating a mental map for how this evaluates.

File: rules/gcp_audit_rules/gcp_destructive_queries.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        "job",
+        "jobConfig",
+        "queryConfig",
+        "destinationTable",
+    ) or deep_get(event, "protopayload", "metadata", "resourceName", default="<TABLE_NOT_FOUND>")
+    return f"GCP: [{actor}] performed a destructive BigQuery [{statement}] query on [{table}]."
+
+
+def alert_context(event):
+    return {
```
Comment (edyesed) [inline rules/gcp_audit_rules/gcp_destructive_queries.py:67]: I would recommend extending alert_context with `actor`, `statement`, and `table` as used in the title.

**Review by edyesed — APPROVED** [review]
:shipit: adds some gcp rules relating some of the audit logged events for bigquery. It looks like bigquery is billed at ~ $5/TB queried, though that's on the monthly aggregate of queried data.

**Review by andrea-youwakim — APPROVED** [review]

**Pull request merged**